src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -15,8 +16,12 @@ internal static partial class DocumentExtensions
         public static bool IsFromPrimaryBranch(this Document document)
             => document.Project.Solution.BranchId == document.Project.Solution.Workspace.PrimaryBranchId;
 
-        public static ValueTask<SyntaxTreeIndex?> GetSyntaxTreeIndexAsync(this Document document, CancellationToken cancellationToken)
-            => SyntaxTreeIndex.GetIndexAsync(document, loadOnly: false, cancellationToken);
+        public static async ValueTask<SyntaxTreeIndex> GetSyntaxTreeIndexAsync(this Document document, CancellationToken cancellationToken)
```

Comment:
If this method can throw null, would a clearer naming convention possibly be something like `GetRequiredSyntaxTreeAsync`?

Reply:
this method shouldn't ever throw.  that's somewhat the point.  however, our type system can't express that with how we pass loadOnly:false inside.  If we had a GetSyntxtreeIndex that returned null (externally) i would def agree with that pattern here. :)